srml/elections-phragmen/src/lib.rs

```diff
@@ -0,0 +1,957 @@
+					locked_balance,
+					T::BlockNumber::max_value(),
+					WithdrawReasons::except(WithdrawReason::TransactionPayment),
+				);
+				<StakeOf<T>>::insert(&who, locked_balance);
+			}
+
+			let now = Self::election_rounds();
+			<VotesOf<T>>::insert(&who, (votes, now));
+		}
+
+		/// Remove `origin` as a voter. One can use this to
+		///   1. _undo_ their votes before the election has started.
+		///   2. Remove their lock and unreserve the bond after an election round.
+		#[weight = SimpleDispatchInfo::FixedNormal(500_000)]
+		fn remove_voter(origin) {
+			let who = ensure_signed(origin)?;
+
+			ensure!(
+				Self::is_current_voter(&who) || <StakeOf<T>>::exists(&who),
+				"should be a current/old voter"
+			);
+
+			// remove storage.
+			<VotesOf<T>>::remove(&who);
+			<StakeOf<T>>::remove(&who);
+
+			// unreserve the bond and remove.
+			T::Currency::unreserve(&who, T::VotingBond::get());
+			T::Currency::remove_lock(MODULE_ID, &who);
+		}
+
+		/// Submit oneself for candidacy.
+		///
+		/// A candidate will either:
+		///   - Lose at the end of the term and gets their bond unreserved
+		///   - Win and become a member. Members will get their bond back at the end ot their term.
+		#[weight = SimpleDispatchInfo::FixedNormal(1_000_000)]
+		fn submit_candidacy(origin) {
+			let who = ensure_signed(origin)?;
+
+			let is_candidate = Self::is_candidate(&who);
+			ensure!(!is_candidate.is_ok(), "duplicate candidate submission");
+			// assured to be an error, error always contains the index.
+			let index = is_candidate.unwrap_err();
+
+			T::Currency::reserve(&who, T::CandidacyBond::get())
+				.map_err(|_| "candidate has not enough funds")?;
+
+			<Candidates<T>>::mutate(|c| c.insert(index, who));
+			CandidateCount::mutate(|c| *c += 1);
+		}
+
+		/// Set the desired member count; if lower than the current count, then seats will not be up
+		/// election when they expire. If more, then a new vote will be started if one is not
+		/// already in progress.
+		#[weight = SimpleDispatchInfo::FixedOperational(10_000)]
+		fn set_desired_seats(origin, #[compact] count: u32) {
+			ensure_root(origin)?;
+			DesiredSeats::put(count);
+		}
+
+		/// Remove a particular member from the set. This is effective immediately.
+		///
+		/// Note: A tally should happen instantly (if not already in a presentation
+		/// period) to fill the seat if removal means that the desired members are not met.
+		#[weight = SimpleDispatchInfo::FixedOperational(10_000)]
+		fn remove_member(origin, who: <T::Lookup as StaticLookup>::Source) {
+			ensure_root(origin)?;
+			let who = T::Lookup::lookup(who)?;
+
+			if Self::is_candidate(&who).is_ok() {
+				let members = Self::members();
+				let new_members: Vec<T::AccountId> = members
+					.into_iter()
+					.filter(|i| *i != who)
+					.collect();
+				<Members<T>>::put(new_members.clone());
+				T::Currency::unreserve(&who, T::CandidacyBond::get());
+				T::ChangeMembers::change_members(&[], &[who], new_members);
+			}
```

Comment:
this doesn't seem to automatically kick-off a new election. it should.

Reply:
okay now I see the reason for the comment above: will start a new election on this call.

Reply:
Okay but wait: 

I assume the purpose of this is to be able to 1) kick out one 2) but keep the rest 3) and find a replacement. This is a bit tricky to do with the current system because we don't store any old vote/candidacy info. 

Assume term durations is 5, [1,2] are elected at block 10. as mentioned, all voters and cadidates are removed from before block 10. at block 13 we kick 2. Now, based on what voting data and candidacy should we start a new election? 

I see two possible answers:
1- the ongoing votes and candidacy that happened in block 10 until 13. In this case, removing one member and starting a new election is moot and this should just be a function named `force_new_election(root)`.
2- elect _only one new member_ from ongoing votes and candidacies in block range [10, 13] (if any) and combine it with the remainder of the previous set.

Reply:
we should store all candidate information except that which is culled.

Reply:
^^^ Is this the case yet @kianenigma ?